### ISSUE TITLE
updating latest examples for using GPU in SPARK EMR 

### ISSUE THREAD

**Issue by gourav-sg** [NVIDIA/spark-rapids#629]
Create new documentation  here https://github.com/NVIDIA/spark-xgboost-examples 

**Location of incorrect documentation**
https://aws.amazon.com/blogs/big-data/improving-rapids-xgboost-performance-and-reducing-costs-with-amazon-emr-running-amazon-ec2-g4-instances/?nc1=b_rp is based on old implementation and is more of a confusion than resolution. Users using xgboost will want to use SQL to pre process data as well. 

**Describe the problems or issues found in the documentation**
See the description here https://github.com/NVIDIA/spark-rapids/issues/628 saying how the documentation is not latest

**Steps taken to verify documentation is incorrect**
https://github.com/NVIDIA/spark-rapids/issues/628

**Suggested fix for documentation**
this is deprecated, and we will need to create new documentation.

---

## Report needed documentation

**Report needed documentation**
Documentation showing that how SPARK should be used in EMR

**Describe the documentation you'd like**
to show how to use GPU clusters for SPARK SQL

**Steps taken to search for needed documentation**
https://github.com/NVIDIA/spark-rapids/issues/628

**Comment by jlowe** [comment]
@gourav-sg can you create this issue in the spark-xgboost-examples repo?  That's where I think the new getting started guide notebook for xgboost would be expected.

**Comment by gourav-sg** [comment]
This has nothing to do with xgboost, this request is for documentation on using SPARK SQL with GPU. What I was pointing out is that the authors should kindly note that while having documentation on xgboost looks great, one should also focus on creating documentation on 80% of data science, which largely includes running SPARK SQL for data preparation/ feature engineering.

**Comment by jlowe** [comment]
The headline of the request mentions updating latest examples, and the first line in the description of the issue is: `Create new documentation here https://github.com/NVIDIA/spark-xgboost-examples`.  If that is the intent of this request then it belongs in the spark-xgboost-examples repo, not here.

If it isn't that and does belong to this project, we need more clarity on exactly what is expected to be done in this repo to resolve this issue.

**Comment by gourav-sg** [comment]
Hi,

We need to have updated notebooks which tells us how to use SPARK SQL with
GPU, this has nothing to do with xgboost, as its applications comes after
the data/ features have been prepared for which most of the users will be
using SPARK SQL.

I am primarily focused on EMR, in AWS, and was working with the community
until wee hours of early morning to figure out the gaps which are not
documented, of presented in outdated and therefore misleading manner in
other places.

So we should have a simple notebook which should show how to start a EMR
cluster bootstrap it with proper CUDA packages, and then run SPARK in it.
Basically an end to end example where people will be able to follow the
instructions and start using SPARK over GPU in EMR.

Currently the fantastic community in NVIDIA has given those packages, but
due to lack of documentation or misleading documentation getting it to use
is a herculean task.


Thanks and Regards,
Gourav Sengupta

**Comment by jlowe** [comment]
Thanks @gourav-sg.  I'm assuming you want a new notebook for AWS EMR under `docs/demo/` then?

If so, I'd like to get the summary and description updated which refers to https://github.com/NVIDIA/spark-xgboost-examples and an Amazon blog post that is not about this repository, leading to a lot of confusion about what's proposed in this issue.

**Comment by mgzhao** [comment]
@gourav-sg, thanks for reporting the issue and review both the AWS blog and our github example code. Your feedback is critical to us.  Like you said, the AWS blog post is for XGBoost GPU accelerated training on Spark 2.x EMR Cluster.   The EMR team has a recent release 6.1.0 including Spark 3.0. We are still working with EMR team to enable GPU Spark 3.0 feature which is mainly the Spark SQL ETL you are interested.  We updated this github site to a new site: https://github.com/mgzhao/spark-xgboost-examples with two branches:  spark-3 and spark-2.  The spark-3 branch does have ETL example code on databrick/standalone.   I am working on publishing a detailed standalone EC2 cluster guide for GPU Spark 3.0.  Once we have GPU Spark 3.0 fully supported on EMR,  we will update you here and also add all the ETL examples/guide on Github.

**Comment by gourav-sg** [comment]
Hi,
thanks a ton for responding back to me. To be honest enough what NVIDIA is
trying to do with SPARK and the rest of the concepts is truly re-building
the entire concept of data to value. You cannot imagine the number of half
minded, brain dead, bespoke, and technically redundant solutions in the
market which are trying to sell themselves off just trying to align
engineering and data science efforts into a single view.

NVIDIA did that with SPARK, and it will be great to start using that. It
should not take more than 8 to 10 hours of work to get an example up and
ready. And I will be eagerly looking ahead for the same.

SPARK 3.x.x was released by Amazon EMR last week, and we should have been
able to start using some of the most exciting and promising features of
SPARK 3 by NVIDIA by now :)

Regards,
Gourav Sengupta
